### PR TITLE
Enhancements to conceal, fold, sign, pmenu

### PR DESCRIPTION
**Pull request by ashwinvis**
FYI. Feel free to close it if you don't like it. And don't ask for screenshots.

### CODE REVIEW

**Comment by ashwinvis** [comment]
Also fixes #12 with a more visible, so far unused brown colour

**Comment by kaicataldo** [comment]
I’m definitely open to changing things and accepting PRs, but it needs to be a collaboration. Since this is entirely about the visual design, screenshots go a long way in helping everyone visualize what these changes are.

Using some of the unused colors (there’s also a magenta) is a great idea.

**Comment by ashwinvis** [comment]
I can do it when I find time - probably not much in the next few months. I just want to let you know how I modified it. You can always pull the branch from my fork and see for yourself.

PS: Also after #19, I am not confident going that extra mile with screenshots is worth it, since you might end up implementing the way you like it.

**Comment by kaicataldo** [comment]
That's disappointing you feel that way.

As far as your last issue goes, I do apologize for the delay. I have been largely inactive on GitHub the last few months for personal reasons, and I work on this project in my free time just like you.

> PS: Also after #19, I am not confident going that extra mile with screenshots is worth it, since you might end up implementing the way you like it.

Since this is a collaborative open source project and we're trying to solve these design problems together, I think it's fair for the maintainer to expect contributors to put in effort to convince others of the changes they are proposing. I'm doing my best to evaluate changes with all end users in mind, and that means that not all requests can be honored. That being said, I don't think this is a fair characterization of what happened. I implemented those changes after you closed your PR with an angry comment.

I'm totally willing to have discussions about changes, but at the end of the day, we have to come to a consensus, and that requires communication and effort on both ends. A great way to avoid throwaway work is to make an issue and reach consensus on the change before making a PR.

As far as screenshots go, I can (and do) pull down branches and test them out before merging, but that's not why I ask for screenshots. They take minimal effort and are not just helpful to me, but allow others who might have an opinion to also chime in.

**Comment by ashwinvis** [comment]
I closed the other PR since I didn't want to deal with the merge conflicts for a simple one-line edit.

> I'm totally willing to have discussions about changes, but at the end of the day, we have to come to a consensus, and that requires communication and effort on both ends. A great way to avoid throwaway work is to make an issue and reach consensus on the change before making a PR.

I had to change the theme anyway, since I use it every day with vim. The changes and regressions affected my workflow. That's why I mentioned I don't mind if you close it, because I will continue to use my fork.

Apologies if I came out as harsh - it was because I was dissappointed. Anyways, I will stop complaining now, as I know it is unfair to have expectations in open-source projects.

> As far as screenshots go, I can (and do) pull down branches and test them out before merging, but that's not why I ask for screenshots. They take minimal effort and are not just helpful to me, but allow others who might have an opinion to also chime in.

So here it goes:

Screenshot_20190702_111559

I checked with all other theme variants and it looks fine as it is now. If any changes are needed (like the magenta suggestion for conceal), I am sure you can edit it yourself right. And perhaps you can post screenshots of what you intend?

**Comment by ashwinvis** [comment]
Folded mode:

Screenshot_20190702_113051

File: colors/material.vim
Comment (kaicataldo) [inline colors/material.vim]: Can this be `call s:SetHighlight('SignColumn', s:fg, s:bg, '')` for consistency? Maybe I'm misunderstanding what this intent of this change is for.

File: colors/material.vim
Comment (kaicataldo) [inline colors/material.vim:201]: I think using `s:brown` is a fantastic idea! Thoughts on also making this `bold` to match the suggested change for `Folded`?
```vim
call s:SetHighlight('Conceal', s:brown, s:bg, '')
```

File: colors/material.vim
Comment (kaicataldo) [inline colors/material.vim]: I like how this looks bolded! My concern with changing it to the same color as comments is that I think it should be differentiated more from the actual text being edited (and to that end, I totally agree that the current color doesn't work). 

I really like the use of brown to highlight `Conceal` characters, and I think it's a fair argument to say that `Folded` could be in the same category (it's a stand in for the actual text). Thoughts on making it brown as well?
```vim
call s:SetHighlight('Folded', s:brown, s:bg, 'bold')
```

Screen Shot 2019-07-02 at 2 47 59 PM

**Review by kaicataldo — CHANGES_REQUESTED** [review]
Thank you, and I really appreciate you sharing where you're coming from. I could have done a better job of explaining why I didn't merge your PR before doing the rewrite. I changed a lot of the colors to match the [theme this is a port of](https://github.com/equinusocio/vsc-material-theme), and the color you had chosen had changed. I wanted to not ignore your request and give you the chance to update it with the new available colors instead of merging your PR and immediately changing it myself.

> If any changes are needed (like the magenta suggestion for conceal), I am sure you can edit it yourself right.

I unfortunately can't! GitHub doesn't let you modify branches belonging to another user's fork without being given explicit permission. If you don't have the time to work on the suggested changes (which I totally understand), I'm happy to make a new PR with the changes, but I want to make sure you get credit here if you'd like it.

I have a few comments/suggestions on the changes.

File: colors/material.vim
Comment (kaicataldo) [inline colors/material.vim]: This is looking a little harsh to me with the black. I like how it is now, but if you feel like it should be differentiated from the background color I also wouldn't be opposed to changing this to
```vim
call s:SetHighlight('Pmenu', s:fg, s:selection, '')
```
Screen Shot 2019-07-02 at 2 49 47 PM

File: colors/material.vim
Comment (ashwinvis) [reply colors/material.vim]: Good idea

File: colors/material.vim
Comment (ashwinvis) [reply colors/material.vim:201]: Conceal should blend in with normal text. For example, latex equation I showed above. Some characters like = are not concealed

File: colors/material.vim
Comment (ashwinvis) [reply colors/material.vim]: ~~I think I tried that and that gives me a gray gutter as shown in a comment in #12.
I use the signify plugin, btw.~~

You are right `s:bg` fixes it.

File: colors/material.vim
Comment (ashwinvis) [reply colors/material.vim]: I use the signify plugin, btw.

File: colors/material.vim
Comment (ashwinvis) [reply colors/material.vim]: Done in 22607db

**Review by kaicataldo — APPROVED** [review]
LGTM. Thanks for contributing.

**Pull request merged**
